backends/candle/src/models/flash_jina_code.rs

```diff
@@ -0,0 +1,458 @@
+            key_layer,
+            value_layer,
+            self.alibi_slopes.as_ref(),
+            cu_seqlens,
+            cu_seqlens,
+            max_s,
+            max_s,
+            self.softmax_scale,
+            false,
+        )?;
+        let attention = attention.flatten_from(candle::D::Minus2)?;
+
+        let hidden_states = self.dense.forward(&attention)?;
+        let hidden_states = self.layer_norm_out.forward(&hidden_states, Some(&residual))?;
+
+        Ok(hidden_states)
+    }
+}
+
+struct JinaBertLayer {
+    attention: AlibiBertAttention,
+    up_gated_layer: Linear,
+    down_layer: Linear,
+    layer_norm_1: LayerNorm,
+    layer_norm_2: LayerNorm,
+    act: HiddenAct,
+
+    intermediate_size: usize,
+
+    span: tracing::Span,
+}
+
+impl JinaBertLayer {
+    pub fn load(vb: VarBuilder, config: &JinaCodeConfig, alibi: Option<Tensor>) -> Result<Self> {
+        let attention = AlibiBertAttention::load(vb.pp("attention"), config, alibi)?;
+
+        let up_gated_weight = vb
+            .pp("mlp")
+            .pp("up_gated_layer")
+            .get((config.intermediate_size * 2, config.hidden_size), "weight")?;
+        let up_gated_layer = Linear::new(up_gated_weight, None, None);
+
+        let down_weight = vb
+            .pp("mlp")
+            .pp("down_layer")
+            .get((config.hidden_size, config.intermediate_size), "weight")?;
+        let down_bias = vb.pp("mlp").pp("down_layer").get(config.hidden_size, "bias")?;
+        let down_layer = Linear::new(down_weight, Some(down_bias), None);
+
+        let layer_norm_1 = LayerNorm::load(
+            vb.pp("layer_norm_1"),
+            config.hidden_size,
+            config.layer_norm_eps as f32,
+        )?;
+        let layer_norm_2 = LayerNorm::load(
+            vb.pp("layer_norm_2"),
+            config.hidden_size,
+            config.layer_norm_eps as f32,
+        )?;
+
+        Ok(Self {
+            attention,
+            up_gated_layer,
+            down_layer,
+            layer_norm_1,
+            layer_norm_2,
+            act: config.hidden_act.clone(),
+            intermediate_size: config.intermediate_size,
+            span: tracing::span!(tracing::Level::TRACE, "layer"),
+        })
+    }
+
+    pub fn forward(
+        &self,
+        hidden_states: &Tensor,
+        cu_seqlens: &Tensor,
+        max_s: usize,
+    ) -> Result<Tensor> {
+        let _enter = self.span.enter();
+
+        let residual = hidden_states.clone();
+        let hidden_states = self.attention.forward(hidden_states, cu_seqlens, max_s)?;
+
+        // Pre-MLP LayerNorm
+        let hidden_states = self.layer_norm_1.forward(&hidden_states, Some(&residual))?;
+
+        // MLP block
+        let residual = hidden_states.clone();
+        let hidden_states = self.up_gated_layer.forward(&hidden_states)?;
+        let non_gated = hidden_states.i((.., .., 0..self.intermediate_size))?;
+        let gated = hidden_states.i((.., .., self.intermediate_size..))?;
```

Comment:
You can use narrow instead:

```suggestion
        let non_gated = hidden_states.narrow(2, 0, self.intermediate_size)?;
        let gated = hidden_states.narrow(2, self.intermediate_size, self.intermediate_size)?;
```

I'm a bit confused, why is hidden_states dim3 here? Shouldn't it be dim2 and narrow on 1?